### DISCUSSION TITLE
Allow jumping to selection in Prompt with letter keys

### DISCUSSION THREAD

**Discussion by tznind** [spectreconsole/spectre.console#992] [Ideas]
**Is your feature request related to a problem? Please describe.**

If `SelectionPrompt` has many entries it would be nice if you could jump to the right area by pressing the corresponding letter.  This is consistent with most other selection systems (e.g. file explorers, listboxes etc).

I have looked in the backlog and not seen this discussed (the closest I found was #172), appologies if I have missed it or if there is an obvious reason why this isn't a good idea. 

**Describe the solution you'd like**
Pressing a letter key (A-Z) should move the selection to the **next** prompt item that begins with that letter

Here is a unit test that demonstrates what I am proposing.  I copied a test in `ListPromptStateTests` for the format

```csharp
[Fact]
public void LetterJumpToSelection()
{
    // Given
    var state = new ListPromptState<string>(
        new[]
        {
            new ListPromptItem<string>("apple"),
            new ListPromptItem<string>("bannana"),
            new ListPromptItem<string>("fish"),
            new ListPromptItem<string>("flamingo"),
        }
        .ToList(), 3, true);

    // First item should be selected
    state.Index.ShouldBe(0);

    // When user presses F
    state.Update(ConsoleKey.F);

    // Then should jump to fish
    state.Index.ShouldBe(2);
            
    // When user presses F again
    state.Update(ConsoleKey.F);

    // Then should jump to flamingo
    state.Index.ShouldBe(3);

    // When user presses F third time
    state.Update(ConsoleKey.F);

    // Then should cycle back to fish
    state.Index.ShouldBe(2);
}
```

I've opened a PR for this here https://github.com/spectreconsole/spectre.console/pull/996

**Comment by chris-crowther** [comment]
This would certainly help with long lists and menus. Perhaps an option to specify a key for each item, so that we could have numbered menus?

**Comment by Frassle** [comment]
I've used other prompt systems where you get a filter text at the top of the prompt. Really useful for long lists.
Could show something like:
```
Filter: [grey]Type to filter choices[/]
▸ elephant
  dog
  cat
  rabbit
```
Then if you press "c" it would filter the list of items to just show "cat". Works for multiselect prompts as well.
